sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1693,7 +1696,9 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
                 // access columns from the source table.
                 val resolvedInsertCondition = insertCondition.map(
                   resolveExpressionByPlanOutput(_, m.sourceTable))
-                val assignments = targetTable.output.map { attr =>
+                val attrs = targetTable.output.filter(targetCol => sourceTable.output.exists(
```

Comment:
can we add a comment to mention where we will handle the missing columns?

Reply:
done